src/high-level/high-level.lisp

```diff
@@ -930,6 +930,30 @@ with upper left block with dimension P-by-Q. Returns the intermediate representa
         (magicl.lapack-cffi::%zgeev jobvl jobvr rows a rows w vl 1 vr rows work lwork rwork info)
         (values (vector-to-list w) (make-matrix :rows rows :cols cols :data vr))))))
 
+(defun hermitian-eig (m)
+  "Finds the (right) eigenvectors and corresponding eigenvalues of a Hermitian square matrix M. Returns as two lists (EIGENVALUES, EIGENVECTORS) where the eigenvalues and eigenvectors are in corresponding order."
+  (let ((rows (matrix-rows m))
+        (cols (matrix-cols m)))
+    (assert (= rows cols) () "M is not a square matrix")
+    (let ((jobz "V")
+          (uplo "U")
+          (n rows)
+          (a (copy-matrix-storage (matrix-data m)))
+          (lda (matrix-rows m))
+          (w (make-D-storage rows))
+          (work (make-Z-storage 1))
+          (lwork -1)
+          (rwork (make-D-storage (- (* 3 rows) 2)))
```

Comment:
should we assert that rows >= 2 above?

Reply:
`RWORK` needs to be of size at least `(max 1 (- (* 3 n) 2))`, but `(plusp n)` will already let me drop the `(max 1 ...)`, and when I added a `plusp` check the compiler told me that it would never execute that code bc it was always true..